### ISSUE TITLE
Sources which are not ready still show as Green Checks

### ISSUE THREAD

**Issue by kingdonb** [weaveworks/vscode-gitops-tools#249]
### Expected behaviour

I have a source which is not ready

The status is as reflected here:

```
status:
  conditions:
  - lastTransitionTime: "2022-04-27T00:03:57Z"
    message: no artifact for resource in storage
    observedGeneration: 1
    reason: NoArtifact
    status: "True"
    type: Reconciling
  - lastTransitionTime: "2022-04-27T00:03:57Z"
    message: 'failed to checkout and determine revision: unable to clone ''https://github.com/kingdonb/podinfo'':
      couldn''t find remote ref "refs/heads/main"'
    observedGeneration: 1
    reason: GitOperationFailed
    status: "False"
    type: Ready
  - lastTransitionTime: "2022-04-27T00:03:57Z"
    message: 'failed to checkout and determine revision: unable to clone ''https://github.com/kingdonb/podinfo'':
      couldn''t find remote ref "refs/heads/main"'
    observedGeneration: 1
    reason: GitOperationFailed
    status: "True"
    type: FetchFailed
  observedGeneration: -1
```

The checkmark should not be green

### Actual behaviour

The source stays green after I refresh the Sources listing

![Screen Shot 2022-04-26 at 8 33 55 PM](https://user-images.githubusercontent.com/3286998/165415179-7087e270-8979-4778-bcee-a5fcc82ef71b.png)

(FWIW, the error with this GitRepository is that it points to a non-existent branch)

### Steps to reproduce

Create a GitRepository like this one:

```
---
apiVersion: source.toolkit.fluxcd.io/v1beta2
kind: GitRepository
metadata:
  name: podinfo
  namespace: flux-system
spec:
  interval: 1m0s
  ref:
    branch: main
  url: https://github.com/kingdonb/podinfo

```

Note that it does not sync. The UI shows the GitRepository with "ready" state, despite refreshing the UI

### Versions

v0.19.2-pre1

**Comment by kingdonb** [comment]
It's much better this time I've tried with #250 

<img width="790" alt="Screen Shot 2022-04-27 at 12 29 08 PM" src="https://user-images.githubusercontent.com/3286998/165567621-538abf7a-83d7-4bd7-8e26-27e69b84301d.png">

**Comment by kingdonb** [comment]
Fixed in v0.19.2-pre4
